pyvista/core/grid.py

```diff
@@ -584,6 +584,7 @@ def __init__(
         spacing=(1.0, 1.0, 1.0),
         origin=(0.0, 0.0, 0.0),
         deep=False,
+        direction_matrix=None,
```

Comment:
We should add explanations for each parameter in the docstring :)

Reply:
whoops! 😅 

Odd that there are no complaints from pre-commit about this (numpydoc or ruff should be linting this kind of thing...)

Reply:
ref #6755

Reply:
> Let's verify if #6755 can catch this error after merging it.

Still not caught by pre-commit, unfortunately :/

Reply:
This might be the issue. This line should be removed:
https://github.com/pyvista/pyvista/blob/ea732e95990b1f3ad56ca1ee3e514142f6d66b59/pyproject.toml#L309

Reply:
Hmm... this isn't it either. That doesn't check for missing members. Plus we define init params in the class docstring, not the __init__ docstring.